app/environments/brassbirmingham/brassbirmingham/envs/test.py

```diff
@@ -143,10 +138,134 @@ def testNetworkSearch(self):
             "Should be",
         )
         p2.buildBuilding(p2.buildings[26], redditch.buildLocations[1])
-        self.assertEqual(p2.money, 2, "Should be")  # 8-5-1
+        self.assertEqual(p2.money, 1, "Should be")  # 8-5-2 (coal market missing 1)
 
         print(board.townDict)
 
+    def testResourceMarketPrice(self):
+        # Empty markets
+        board.coalMarketRemaining = 0
+        assert board.priceForCoal(4) == 4 * 8
+        board.ironMarketRemaining = 0
+        assert board.priceForIron(5) == 5 * 6
+
+        # Single needed
+        board.coalMarketRemaining = 8
+        assert board.priceForCoal(1) == 4
+        board.ironMarketRemaining = 8
+        assert board.priceForIron(1) == 2
+
+        # Single price jump
+        board.coalMarketRemaining = 8
+        assert board.priceForCoal(3) == 13
+        board.ironMarketRemaining = 8
+        assert board.priceForIron(3) == 7
+
+        # Big price jump
+        board.coalMarketRemaining = 13
+        assert board.priceForCoal(10) == 35
+        board.ironMarketRemaining = 8
+        assert board.priceForIron(10) == 40
+
+    def testVictoryPoints(self):
+        # Zero points
+        resetGame(2)
+        playerVPs = board.getVictoryPoints()
```

Comment:
Wouldn't it make more sense to just attach a `vp` attr to the `Player` object?

Reply:
VPs are only incremented at the end of each era and it's effectively a "board-based" calculation, in my opinion which is why I added it to board.

Player objects will store their intermediate VPs from Canal to Rail era though